### PR TITLE
cbtemulator: init at 1.22.0

### PR DESCRIPTION
**Pull request by flokli**
This packages cbtemulator, an in-memory emulator for Google Bigtable.

I added a small smoketest, setting up tables, inserting a value, and asserting we get it back.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by cbrewster — APPROVED** [review]

**Comment by ryantm** [comment]
Tried to build this and got this problem:
```
$ nix run github:nixos/nixpkgs?ref=pull/298044/head#cbtemulator -L
cbtemulator> Running phase: unpackPhase
cbtemulator> unpacking source archive /nix/store/fr9lkc63nviqcwy6s83i1pg53fqffh3h-source
cbtemulator> source root is source
cbtemulator> Running phase: patchPhase
cbtemulator> Running phase: updateAutotoolsGnuConfigScriptsPhase
cbtemulator> Running phase: configurePhase
cbtemulator> Running phase: buildPhase
cbtemulator> go: 'go mod vendor' cannot be run in workspace mode. Run 'go work vendor' to vendor the workspace or set 'GOWORK=off' to exit workspace mode.
error: builder for '/nix/store/l2kx177461hmki26ir5qz327ys71hxql-cbtemulator-1.22.0-go-modules.drv' failed with exit code 1;
```

**Comment by flokli** [comment]
@ryantm I don't quite get it. I first assumed some cached FOD output on my side, removed `modRoot` and `goWork`, and added  `/bigtable` , but it came up with the same `vendorHash`, and builds on my machne (x86_64-linux). Does it still fail for you now?

File: pkgs/by-name/cb/cbtemulator/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+buildGoModule rec {
+  pname = "cbtemulator";
+  version = "1.22.0";
+
+  src = (fetchFromGitHub {
+    owner = "googleapis";
+    repo = "google-cloud-go";
+    rev = "bigtable/v${version}";
+    hash = "sha256-eOi4QFthnmZb5ry/5L7wzr4Fy1pF/H07BzxOnXtmSu4=";
+  }) + "/bigtable";
```
Comment (sikmir) [inline pkgs/by-name/cb/cbtemulator/package.nix:20]: What about  `sourceRoot`?

**Comment by cbrewster** [comment]
It builds for me, I also tried recalculating the `vendorHash` and it matches what is shown in this PR. :thinking:

File: pkgs/by-name/cb/cbtemulator/package.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    # Sets up a table and family, then inserts, and ensures it gets back the value.
+    tests.smoketest = runCommand "cbtemulator-smoketest"
+      {
+        nativeBuildInputs = [ google-cloud-bigtable-tool procps ];
+      } ''
+      # Start the emulator
+      ${lib.getExe pkgs.cbtemulator} &
+
+      cleanup() {
+        pkill cbtemulator
```
Comment (flokli) [inline pkgs/by-name/cb/cbtemulator/package.nix:39]: Note to self: this doesn't seem to bring pkill into $PATH, check Darwin manually

**Comment by ryantm** [comment]
> Does it still fail for you now?

Strange. It builds for me fine now!

**Review by ryantm — APPROVED** [review]

File: pkgs/by-name/cb/cbtemulator/package.nix (hunk 1 above)
Comment (flokli) [reply pkgs/by-name/cb/cbtemulator/package.nix:20]:
```
building '/nix/store/j46p6z7pyvwxzwcc992w5kmsldrw2xmb-cbtemulator-1.22.0-go-modules.drv' on 'ssh://root@box.flokli.de'...
copying 0 paths...
Running phase: unpackPhase
unpacking source archive /nix/store/fr9lkc63nviqcwy6s83i1pg53fqffh3h-source
source root is bigtable
chmod: cannot access 'bigtable': No such file or directory
```

Comment (flokli) [reply pkgs/by-name/cb/cbtemulator/package.nix:20]: I added a comment.

File: pkgs/by-name/cb/cbtemulator/package.nix (hunk 2 above)
Comment (flokli) [reply pkgs/by-name/cb/cbtemulator/package.nix]: `procps` doesn't have `pkill` on MacOS, but I rewrote this to just use the builtin `kill` bash builtin.

**Comment by flokli** [comment]
Fixed the smoketest on aarch64-darwin.

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 298044` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cbtemulator</li>
  </ul>
</details>

File: pkgs/by-name/cb/cbtemulator/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  '';
+
+  passthru = {
+    # Sets up a table and family, then inserts, and ensures it gets back the value.
+    tests.smoketest = runCommand "cbtemulator-smoketest"
+      {
+        nativeBuildInputs = [ google-cloud-bigtable-tool ];
+      } ''
+      # Start the emulator
+      ${lib.getExe pkgs.cbtemulator} &
```
Comment (sikmir) [inline pkgs/by-name/cb/cbtemulator/package.nix:37]:
```suggestion
      ${lib.getExe cbtemulator} &
```

File: pkgs/by-name/cb/cbtemulator/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
+{ pkgs
```
Comment (sikmir) [inline pkgs/by-name/cb/cbtemulator/package.nix:1]:
```suggestion
{ cbtemulator
```

File: pkgs/by-name/cb/cbtemulator/package.nix (hunk 3 above)
Comment (flokli) [reply pkgs/by-name/cb/cbtemulator/package.nix]: Done

File: pkgs/by-name/cb/cbtemulator/package.nix (hunk 4 above)
Comment (flokli) [reply pkgs/by-name/cb/cbtemulator/package.nix]: Done

**Pull request merged**